src/app/interfaces/strongRoute.ts

```diff
@@ -142,7 +159,8 @@ export class StrongRoute {
    * Router representation of the route
    */
   toRoute(): string[] {
-    return this.full.map(x => x.name).filter(x => !!x);
+    const output = this.full.map(x => x.name);
+    return output.length > 1 ? output.filter(x => !!x) : [""];
```

Comment:
what does this condition guard against? why do we need to return an array with an empty string over an empty array? I'd say an empty array is more semantically correct.

Reply:
`[""]` is the base route. I remove it during the filter because otherwise you would have `['""', 'projects']` which would not be a valid route. As a counter-point,  i sometimes need `[""]` as a route (eg the HomeMenuItem), so i need `[""]` as an output. The only time its the output is if it is the only route in the list.

Reply:
Where does a case like this `['""', 'projects']` happen? As I read the code, it shouldn't.

The first component should always be `null` in  `StrongRoute.Base.add("projects")`

So why not just slice?

And for home? Rather than `null` the first component should be `""`. E.g.   `StrongRoute.Base.add("")`

Then there's always a null and code simply becomes: `this.slice(1).map(x => x.name)`